go/test/endtoend/vtgate/foreignkey/fk_fuzz_test.go

```diff
@@ -703,14 +705,74 @@ func TestFkFuzzTest(t *testing.T) {
 	}
 }
 
-func validateReplication(t *testing.T) {
-	for _, keyspace := range clusterInstance.Keyspaces {
-		for _, shard := range keyspace.Shards {
-			for _, vttablet := range shard.Vttablets {
-				if vttablet.Type != "primary" {
-					checkReplicationHealthy(t, vttablet)
-				}
-			}
+// BenchmarkFkFuzz benchmarks the performance of Vitess unmanaged, Vitess managed and vanilla MySQL performance on a given set of queries generated by the fuzzer.
+func BenchmarkFkFuzz(b *testing.B) {
+	maxValForCol := 10
+	maxValForId := 10
+	insertShare := 50
+	deleteShare := 50
+	updateShare := 50
+	numQueries := 1000
+	// Wait for schema-tracking to be complete.
+	waitForSchemaTrackingForFkTables(b)
+	for i := 0; i < b.N; i++ {
+		// Clear out all the data to ensure we start with a clean slate.
+		startBenchmark(b)
+		// Create a fuzzer to generate and store a certain set of queries.
+		fz := newFuzzer(1, maxValForId, maxValForCol, insertShare, deleteShare, updateShare, SQLQueries, nil)
+		var queries []string
+		for j := 0; j < numQueries; j++ {
+			genQueries := fz.generateQuery()
+			require.Len(b, genQueries, 1)
+			queries = append(queries, genQueries[0])
+		}
+
+		// Connect to MySQL and run all the queries
+		mysqlConn, err := mysql.Connect(context.Background(), &mysqlParams)
+		require.NoError(b, err)
+		// Connect to Vitess managed foreign keys keyspace
+		vtConn, err := mysql.Connect(context.Background(), &vtParams)
```

Comment:
nit: mcmp can be used here to offer a shared struct for both connections

Reply:
Yeah, I thought of that too, but mcmp is using `t *testing.T` inside and it uses it to mark all its functions as `t.Helper()`. This is not something we can do for benchmarks. Essentially we'll have to change mcmp to take in `t testing.TB` and then do a type check for `*testing.T` before calling `t.Helper()`. It didn't feel worth it.